core/src/main/java/org/elasticsearch/cluster/ClusterName.java

```diff
@@ -33,6 +33,8 @@
     public static final Setting<ClusterName> CLUSTER_NAME_SETTING = new Setting<>("cluster.name", "elasticsearch", (s) -> {
         if (s.isEmpty()) {
             throw new IllegalArgumentException("[cluster.name] must not be empty");
+        } else if(s.contains(":")){
+            throw new IllegalArgumentException("[cluster.name] must not contains ':' character");
```

Comment:
Duplicating the string is fine, the maintenance here is low as this string is not going to be changing, and the lack of indirection keeps it simple.